neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/query/parsing/impl/Neo4jAliasResolver.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.neo4j.query.parsing.impl;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.hibernate.AssertionFailure;
+import org.hibernate.ogm.datastore.neo4j.dialect.impl.NodeLabel;
+import org.hibernate.ogm.util.impl.StringHelper;
+
+/**
+ * Keep track of the aliases needed to create the Cypher query.
+ *
+ * @author Davide D'Alto
+ * @author Guillaume Smet
+ */
+public class Neo4jAliasResolver {
```

Comment:
I didn't forget but looks like a rebase issue. Will fix.

The name was too general IMHO and I wanted to make it clear it was a Neo4j specific thing as it's based on the Neo4j relationships.